react/ProductContextProvider.tsx

```diff
@@ -4,6 +4,76 @@ import ProductContext from './ProductContext'
 import { ProductDispatchContext } from './ProductDispatchContext'
 import { useProductReducer, getSelectedItem } from './reducer'
 import { getSelectedSKUFromQueryString } from './modules/skuQueryString'
+import { MaybeProduct, Item } from './ProductTypes'
+
+export interface ProductAndQuery {
+  query: Record<string, any>
+  product: MaybeProduct
+}
+
+type GroupId = string
+
+export interface AssemblyOptionItem {
+  id: string
+  quantity: number
+  seller: string
+  initialQuantity: number
+  choiceType: string
+  name: string
+  price: number
+  children: Record<string, AssemblyOptionItem[]> | null
+}
+
+export interface BuyButtonContextState {
+  clicked: boolean
+}
+
+export interface ProductContextState {
+  selectedItem?: Item | null
+  product: MaybeProduct
+  selectedQuantity: number
+  skuSelector: {
+    selectedImageVariationSKU: string | null
+    isVisible: boolean
+    areAllVariationsSelected: boolean
+  }
```

Comment:
Maybe these attributes of `skuSelector` should be partial too, just for safity

Reply:
Ok! No problem 😄